sensu/plugins/check-dir-new-files.rb

```diff
@@ -0,0 +1,98 @@
+#! /usr/bin/env ruby
+#
+#   check-dir-new-files
+#
+# DESCRIPTION:
+#   Checks the number of specific files in a directory
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux, BSD
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#   #YELLOW
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2014 Sonian, Inc. and contributors. <support@sensuapp.org>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'fileutils'
+require 'time'
+
+class DirCount < Sensu::Plugin::Check::CLI
+  BASE_DIR = '/var/cache/check-dir-new-files'
```

Comment:
I will update the directory as you suggested. 
For this check, it need to have permission to access both the target directory and state_file directory, so still need sudo.